pkg/client/outofband/client.go

```diff
@@ -331,25 +335,42 @@ func didServiceBlockFunc(p Provider) func() (*did.Service, error) {
 
 		verKeyB58 := base58.Encode(verKey)
 
-		// get the route configs
-		serviceEndpoint, routingKeys, err := mediator.GetRouterConfig(routeSvc, p.ServiceEndpoint())
+		connections, err := routeSvc.GetConnections()
 		if err != nil {
-			return nil, fmt.Errorf("didServiceBlockFunc: create invitation - fetch router config : %w", err)
+			return nil, fmt.Errorf("didServiceBlockFunc: create invitation - fetch router connections: %w", err)
 		}
 
-		svc := &did.Service{
-			ID:              uuid.New().String(),
-			Type:            "did-communication",
-			Priority:        0,
-			RecipientKeys:   []string{verKeyB58},
-			RoutingKeys:     routingKeys,
-			ServiceEndpoint: serviceEndpoint,
+		if len(connections) == 0 {
+			return []*did.Service{{
+				ID:              uuid.New().String(),
+				Type:            "did-communication",
+				RecipientKeys:   []string{verKeyB58},
+				ServiceEndpoint: p.ServiceEndpoint(),
+			}}, nil
 		}
 
-		if err = mediator.AddKeyToRouter(routeSvc, base58.Encode(verKey)); err != nil {
-			return nil, fmt.Errorf("didServiceBlockFunc: create invitation - failed to add key to the router : %w", err)
+		var services []*did.Service
+
+		for i := range connections {
+			// get the route configs
+			serviceEndpoint, routingKeys, err := mediator.GetRouterConfig(routeSvc, connections[i], p.ServiceEndpoint())
+			if err != nil {
+				return nil, fmt.Errorf("didServiceBlockFunc: create invitation - fetch router config : %w", err)
+			}
+
+			if err = mediator.AddKeyToRouter(routeSvc, connections[i], base58.Encode(verKey)); err != nil {
```

Comment:
This strategy is hard-coded and is registering the one key created for this OOB invitation/request on all registered routers. This must be discussed. I don't believe this is the correct approach.

A better approach in my mind would be for us to inject the router registration algorithm as a strategy.

Reply:
I see that current logic leads us to `chooseTarget` function. It means that we always pick up the first `service`.
Which makes current logic (`loop`) redundant. 
As I understand we can only use one service for `invitation`. 
I will rollback this change. Also, I will provide a possibility to inject `didDocSvcFunc` function. 
Which as I understand will play `router registration algorithm` role.
@llorllale Does it make sense?

Reply:
Done! Added possibly to provide the router`s connection.